system/BaseModel.php

```diff
@@ -1090,12 +1090,19 @@ public function onlyDeleted()
      */
     public function replace(array $data = null, bool $returnSQL = false)
     {
+        $data = $data ?? [];
```

Comment:
Sorry, I messed up the reference on mobile...
I was thinking about this line. I do not really understand why it is needed here.

If I'm correct, the null coalescing operator on `$data` will only reset it to `[]` if `$data` is `null``, as it is typehinted.
In line 1096, we do `if($data && ...  && validate($data))`, both `[]` and `null` would evaluate `false` in the first condition, so we do not need to enure `$data` to be array here.

In the second `if` block, `array_key_exists`  requires an array, so I think we can pull the null coalescence directly before the second `if`. It is not needed in the first `if`.

I hope you can understand what I mean 😆

Reply:
Ah yes. It's misplaced. I'll fix that when I'm not on mobile. Thanks!